### ISSUE TITLE
Mojaloop Helm release v16.1.0

### ISSUE THREAD

**Issue by elnyry-sam-k** [mojaloop/project#3847]
## **Goal**:
`As a` ML OSS adopter

`I want to` have a Mojaloop release v16.1.0 (version to be confirmed as part of the story) that is successfully validated by the Golden path tests
  
`so that` I can adopt and use a version of Mojaloop Switch services bundle that is tested and contains the latest features enhancements and bug fixes added in PI-23, 24 (includes Merchant payments, performance enhancements done post ML v16.0.0 - Congo and several security patches)

**Acceptance Criteria**:
- [ ] Update "APP VERSION" for each chart.yaml
- [ ] Helm RC Release is validated on Moja2 to achieve 100% result with TTK GoldenPath 
- [ ] Helm RC Release is validated with PISP provisioning and test collection (TTK) with 100%
- [ ] Helm RC Release is validated with Bulk test collection (TTK) with 100%
- [ ] QA (Golden path) on deployments done on AWS Moja2 environment is done
- [ ] Bugs are logged based on observations, failures
- [ ] Bugs logged are fixed if critical to make subsequent releases in individual services (or Helm)
- [ ] All the changes needed for supporting the Helm Release are done
- [ ] Helm chart is published
- [ ] Release notes are published
- [ ] Test cases are published (TTK)
- [ ] TTK tests run successfully as Post-Hook jobs and Tests

**Notes**:
- 

**Complexity**: Medium - Involves all core and supporting services
**Uncertainty**: Medium

**Tasks**:
- [ ] Deploy, validate RC on AWS Moja2 
  - [ ] Validate merchant test collection works as expected
- [ ] Identify issues with QA Scripts if any - fix them and retest 
- [ ] QA for bugs, regressions, log them 
- [ ] Fix bugs logged if critical 
- [ ] Validate with TTK GP collection 
- [ ] TTK test-cases repo Release is published 
- [ ] Release notes for helm RC is drafted
  - [ ] Guidance for migration from Release (may have to be a separate story if sufficiently big / complex)
- [ ] Update release notes with Upgrade Strategy Guide link
- [ ] Helm Release is published 
- [ ] Deployment of released Helm on **Moja2** 
  - [ ] Helm release is deployed on **Moja2** successfully
  - [ ] Regression testing on Moja2 using TTK collections 
    - [ ] GP collection
    - [ ] Core Bulk collection
    - [ ] Third-party collection
    - [ ] SDK Bulk collection
    - [ ] SDK R2P collection
    - [ ] Merchant payment test collection
  - [ ] Validation using CGS Golden path regression tests etc on Moja2 
  - [ ] Test upgradability from previous release
- [ ] Deployment of released Helm on **Moja1** 
  - [ ] Helm release is deployed on **Moja1** successfully
  - [ ] Validation using Golden path regression tests etc on Moja1 
    - [ ] GP collection
    - [ ] Core Bulk collection
    - [ ] Third-party collection
    - [ ] SDK Bulk collection
- [ ] Merchant payment test collection
  - [ ] Validation using CGS Golden path regression tests etc on Moja1 
  - [ ] Validate daily cronJobs on Moja1 for GP and Cleanup scripts 

**Done**
- [ ] Acceptance Criteria pass
- [ ] Designs are up-to date
- [ ] TTK Tests pass
- [ ] Code Style & Coverage meets standards
- [ ] Release notes are verified

## **Pull Requests**:
- [ ] 

## **Issues**:
- [ ] 

## **Follow-up**:

**Dependencies**:
- N/A

## **Accountability**:
- Owner: 
- QA/Review: @elnyry-sam-k @vijayg10 @oderayi

**Comment by elnyry-sam-k** [comment]
Updated status as of 12th March (depending on the timezone :slightly_smiling_face: ) based on the latest run of the test collections:

Grouping these into three categories - the first one is good and we want everything here in this group to go for the release; #2 is the list of items where we have a fix in the pipeline and these need a new helm dev release which will then be retested (after redeployment) and the third group is the list of current pending issues (to be triaged, debugged and fixed as necessary) .. 

1. Completed:
1.1. Dev release - default mode - GP tests passing 100%
1.2 Dev release - default mode - Interscheme tests passing 100%
1.3 Dev release - iso20022 mode enabled - GP tests passing 100%
1.4 Dev release - default mode - config change: third-party collection - passing 100%
1.5. Dev release - “on-us” transfers passing 100%
1.6 Dev release -  “quoting service pass-through mode disabled” passing 100%
1.7 Dev release - “batching disabled” fails (550 failures) : (Will not be required after the latest fix to have batching as native) - will be N/A from future updates
2. Waiting on the fix to be merged in to main, then will be redeployed and tested:
2.1. Dev release - default mode - config change: bulk collection failing - (Fix made by Steven, to be merged and retested)
2.2. Dev release - default mode - RequestToPay in SDK (not switch, triggered by backend through SDK) - (Fix made by Steven, to be merged and retested)
3. Pending failures to be addressed:
3.1 Dev release - CGS - 1 assertion failure (new issue) 
3.2 Dev release - default mode - config change: Interscheme disabled -  11 failures ( transfer fails saying ipl and condition do not match with quotes, quote fails)

**Comment by elnyry-sam-k** [comment]
Dev release details (with example commands for helm deploy) used which provided above results:

For (example) backend:
`helm --namespace moja3 install backend mojaloop/example-mojaloop-backend --devel --version 16.0.4-20869-5510a51`

For mojaloop deployment:
`helm --namespace moja3 install moja mojaloop/mojaloop --devel --version 16.0.4-20869-5510a51`

**Comment by elnyry-sam-k** [comment]
Test results run on v17.0.0 on a sample dev environment, attached for reference:

1. GP collection (FX  included): [GP test results run on 27-Mar-2025](https://github.com/user-attachments/files/19510202/GP.passing.zip)
2. GP ISO 20022 collection: [ISO test results run on 27-Mar-2025](https://github.com/user-attachments/files/19510286/ISO.20022.Passing.zip)
3. Core Bulk collection: [Core Bulk test results run on 27-Mar-2025](https://github.com/user-attachments/files/19510258/Bulk.Passing.zip)
4. Third-party collection: [Third-party test results run on 28-Mar-2025](https://github.com/user-attachments/files/19510264/Third.Party.Passing.zip)
5. SDK Bulk collection: [SDK Bulk Passing test results run on 28-Mar-2025.zip](https://github.com/user-attachments/files/19510351/SDK.Bulk.Passing.zip)
6. SDK R2P collection: [Request To Pay test results run on 28-Mar-2025.zip](https://github.com/user-attachments/files/19510347/Request.To.Pay.zip)
7. Interscheme test collection: [Interscheme test results run on 28-Mar-2025](https://github.com/user-attachments/files/19510114/ml-v1700-interscheme-28mar2025.zip)
